### ISSUE TITLE
On the phase diagram of Spin-1 cases

### ISSUE THREAD

**Issue by JerryChen97** [JerryChen97/KitaevLadder#31]
I spent some days improving the phase diagram of spin-1 cases but there has been no significant change.
![image](https://user-images.githubusercontent.com/21050723/89209494-654e7600-d58c-11ea-9c60-71b64a8d80ce.png)
Here is a diagram just as usual, where we calculate the correlation length (logarithm) for each phase point at two different bond dimensions 64 and 128.

The main part is still composed of gapped points that have no change in the correlation length, which is fine; but for certain points gathered in the center of two diagonal directions, they seemed gapless but also sometimes calculated inaccurately, which seems quite creepy.

Then, I look closer into the line `Jx=Jy`, do the DMRG in a series (iteratively using the previous `psi` as the input of the next DMRG engine), and here is an interesting figure for the correlation length (the original value, not log)
![image](https://user-images.githubusercontent.com/21050723/89210047-69c75e80-d58d-11ea-939f-7e361bde6a92.png)

The two peak points correspond to `Jx=Jy=.84` and `Jx=Jy=1.40` respectively.

**Comment by JerryChen97** [comment]
Here is a fig for the finite-scaling of entanglement entropy w.r.t. the log of correlation length at `Jx=Jy=1.4, Jz=1`
![image](https://user-images.githubusercontent.com/21050723/89210215-b448db00-d58d-11ea-8500-b730976ad733.png)

**Comment by JerryChen97** [comment]
> Here is a fig for the finite-scaling of entanglement entropy w.r.t. the log of correlation length at `Jx=Jy=1.4, Jz=1`
> ![image](https://user-images.githubusercontent.com/21050723/89210215-b448db00-d58d-11ea-8500-b730976ad733.png)

A fitting gives the central charge to be `0.5` here.

**Comment by JerryChen97** [comment]
> > Here is a fig for the finite-scaling of entanglement entropy w.r.t. the log of correlation length at `Jx=Jy=1.4, Jz=1`
> > ![image](https://user-images.githubusercontent.com/21050723/89210215-b448db00-d58d-11ea-8500-b730976ad733.png)
> 
> A fitting gives the central charge to be `0.5` here.

`chi_list = [16, 32, 64, 128, 256]`

**Comment by JerryChen97** [comment]
The `.84` point seems a little bit not so typical. I will try finding a 'more gapless' point.
![image](https://user-images.githubusercontent.com/21050723/89225940-18c56380-d5a9-11ea-917d-781f39c00a65.png)

**Comment by JerryChen97** [comment]
Here is a rough phase diagram given by directly calculating the correlation length of each point.
![image](https://user-images.githubusercontent.com/21050723/89349578-6232b300-d67c-11ea-9649-6311069edf47.png)

It seems that there is a rocket-shaped gapless circle lying in the center.

**Comment by JerryChen97** [comment]
![image](https://user-images.githubusercontent.com/21050723/89684049-a960b400-d8c7-11ea-9288-2c91520f2627.png)

Here is a much more precise phase diagram given by `log xi` (all the log below zero were trimmed to be just zero). @aaronszasz How do you like it?

**Comment by JerryChen97** [comment]
Here is the original version:
![image](https://user-images.githubusercontent.com/21050723/89684478-8b478380-d8c8-11ea-998a-444d6f24d931.png)

**Comment by JerryChen97** [comment]
I got a feeling that maybe the real phase diagram is much more complicated and elegant than I've thought

**Comment by aaronszasz** [comment]
This looks really interesting!  So it looks like there are really three phases, maybe?

**Comment by JerryChen97** [comment]
> This looks really interesting! So it looks like there are really three phases, maybe?

Yeah that's my current guess.

**Comment by JerryChen97** [comment]
And the shape of the left-below region seems really unusual; also I am curious about what happened to the border between the left-below region and the central region.

**Comment by JerryChen97** [comment]
Some other figs along the line `Jx=Jy`, where the resolution is 0.01
![image](https://user-images.githubusercontent.com/21050723/90677968-14957900-e22c-11ea-9161-5a3ad00c4b48.png)

![image](https://user-images.githubusercontent.com/21050723/90677980-195a2d00-e22c-11ea-940b-9c7fb931b380.png)

![image](https://user-images.githubusercontent.com/21050723/90678029-29720c80-e22c-11ea-8c5d-0e07a8851b68.png)

**Comment by JerryChen97** [comment]
Next: fDMRG for the critical points

**Comment by JerryChen97** [comment]
Here is a map for XY local order parameter using the same dataset of spin-1 case
![image](https://user-images.githubusercontent.com/21050723/95135167-08a93c00-0732-11eb-8266-f67f67040a51.png)

**Comment by aaronszasz** [comment]
Excellent, so that's one phase identified!

**Comment by aaronszasz** [comment]
I suspect that one of the two remaining phases will correspond to the other spin-1/2 phase, and the last one will be new, but we'll see

**Comment by JerryChen97** [comment]
I just got the diagrams for the `X`-string order parameter (numerically calculated by `<X_1 ... X_51>`) and the `Z`-string order parameter (numerically calculated by `<Z_0 ... Z_50>`) using the same dataset.
![image](https://user-images.githubusercontent.com/21050723/95240995-2f6f7d00-07db-11eb-811b-5aae3a50ce45.png)
![image](https://user-images.githubusercontent.com/21050723/95241008-34ccc780-07db-11eb-95b6-7ad3b842770c.png)

**Comment by JerryChen97** [comment]
PS: only the right-below triangular part of the diagram was really computed; the other half was produced by simply flipping the same values along the diagonal line `Jx=Jy`. Therefore, in the left-upper part the real op should be the `Y`-string order parameter

**Comment by JerryChen97** [comment]
@aaronszasz I got a feeling that something novel happened here when the system was changed from spin-1/2 to spin-1: in spin-1/2 case, both `XY` and `Z`-string can indicate the `Z`-phase, and `X`-string indicates the `X`-phase; however, in spin-1 case, if my iDMRG data are correct enough, the `XY` order parameter and `Z`-string order parameter are 'taken apart' from each other.

**Comment by JerryChen97** [comment]
In theory, do we have any technique to map the physical spin from discrete values to continuous values? I heard from someone that there is a way to transform spin models to another set of models, which implements such stuff, but right now I couldn't recall the details....

**Comment by aaronszasz** [comment]
These string order parameters for spin 1 look to me like they are not providing any information.  Rather, there is a tendency when Jz is large to orient along Z, likewise for Jx and X.  This is why these order parameters seem to decay to 0 at random places rather than the phase boundary.

Can you show the equivalent of these figures for spin-1/2?

**Comment by JerryChen97** [comment]
Spin 1/2 case:
![image](https://user-images.githubusercontent.com/21050723/95247992-9c3b4500-07e4-11eb-9263-f21b0ca2d896.png)
![image](https://user-images.githubusercontent.com/21050723/95248009-a2c9bc80-07e4-11eb-84e7-d5d53a3fbbf2.png)

**Comment by JerryChen97** [comment]
> Spin 1/2 case:
> ![image](https://user-images.githubusercontent.com/21050723/95247992-9c3b4500-07e4-11eb-9263-f21b0ca2d896.png)
> ![image](https://user-images.githubusercontent.com/21050723/95248009-a2c9bc80-07e4-11eb-84e7-d5d53a3fbbf2.png)

These diagrams were created by multiply only 10 ops together; I didn't include more because TeNPy didn't provide 'Sigma' operators for the `SpinSite` model and this made me multiply an extra `2**n` where `n` is the number of spin operators, and if too many operators are considered I think the final result will be kinda spoiled.
This choice made the results not to perfectly reflect the precise theoretical prediction.

In theory, as the number of ops approaches infinity, `X`-string parameter immediately vanishes in the `Z` region and vice versa for `Z`-string.

**Comment by JerryChen97** [comment]
There are also some diagrams I created before that might be useful, in [another issue](https://github.com/JerryChen97/KitaevLadder/issues/34#issuecomment-685795400)

**Comment by aaronszasz** [comment]
This looks very different from the spin 1 case, to me.  The "boundary" of the string order parameter is exactly at the phase boundary here.

**Comment by JerryChen97** [comment]
> This looks very different from the spin 1 case, to me. The "boundary" of the string order parameter is exactly at the phase boundary here.

I agree. The boundary lines can still be observed somehow but in quite an unclear way, especially for `Z`-string which only provides a diff in values instead of whether or not to vanish. I am not sure whether this is caused by the finite length of string operators I computed or it's something more fundamental.

**Comment by aaronszasz** [comment]
It appears from the figure you posted in the other issue that even if the string expectation value does not vanish at the phase transition, it would have infinite slope at that point in the limit where the string length goes to infinity, ie there is a singularity there.  You can possibly check this by extrapolation up to ~20 spins?  I'm not sure of the physical significance of that, precisely, but it is again distinct from the spin-1 case.
